### PR TITLE
[DO NOT MERGE] Test astro-sdk with airflow 2.4.0rc1

### PR DESCRIPTION
**Pull request by phanikumv**
Dummy PR to test that astro-sdk works well with Airflow 2.4.0rc1

### CODE REVIEW

File: python-sdk/pyproject.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ license = {file = "LICENSE"}
 
 requires-python = ">=3.7"
 dependencies = [
-    "apache-airflow>=2.0",
+    "apache-airflow>=2.4.0rc1",
```
Comment (feluelle) [inline python-sdk/pyproject.toml:20]: Why are we bumping the **minimum** version?

Comment (feluelle) [reply python-sdk/pyproject.toml]: Ah sorry, this is only for testing, right @phanikumv?

Comment (phanikumv) [reply python-sdk/pyproject.toml]: this is to test the code on 2.4.0b1. This PR is just to check whether all the CI unit tests run fine or not. We wont merge it

Comment (phanikumv) [reply python-sdk/pyproject.toml]: thats right @feluelle

File: python-sdk/pyproject.toml [hunk 2]
```diff
@@ -17,7 +17,7 @@ license = {file = "LICENSE"}
 
 requires-python = ">=3.7"
 dependencies = [
-    "apache-airflow>=2.0",
+    "apache-airflow==2.4.0rc1",
```
Comment (kaxil) [inline python-sdk/pyproject.toml:20]:
```suggestion
    "apache-airflow==2.4.0b1",
```

File: python-sdk/noxfile.py [hunk 3]
```diff
@@ -21,7 +21,7 @@ def dev(session: nox.Session) -> None:
 
 
 @nox.session(python=["3.7", "3.8", "3.9"])
-@nox.parametrize("airflow", ["2.2.5", "2.3"])
+@nox.parametrize("airflow", ["2.2.5", "2.3", "2.4.0rc1"])
```
Comment (kaxil) [inline python-sdk/noxfile.py:24]:
```suggestion
@nox.parametrize("airflow", ["2.2.5", "2.3", "2.4.0b1"])
```

File: python-sdk/noxfile.py [hunk 4]
```diff
@@ -21,7 +21,7 @@ def dev(session: nox.Session) -> None:
 
 
 @nox.session(python=["3.7", "3.8", "3.9"])
-@nox.parametrize("airflow", ["2.2.5", "2.3"])
+@nox.parametrize("airflow", ["2.2.5", "2.3", "2.4.0b1"])
```
Comment (kaxil) [inline python-sdk/noxfile.py:24]: I am just trying it with 2.4.0b1 -- and see if we can reproduce the error

Comment (phanikumv) [reply python-sdk/noxfile.py:24]: wierd, because i tried in morning with 2.4.0b1 and didnt get this error

Comment (phanikumv) [reply python-sdk/noxfile.py:24]: but you didnt change `pyproject.toml` ?

**Pull request closed without merging**